### ISSUE TITLE
Pyscannerbit interface - feeding simple function to PS

### ISSUE THREAD

**Issue by ZacSpec** [DarkMachines/high-dimensional-sampling#12]
When the simple function (get_simple_interface()) is passed to Pyscannerbit's scan function, PS uses inspect.getargspec to determine the number of arguments of the function, but it appears to only find one. I've avoided this so far by using 'neg_f = lambda x,y: -simple(x,y)', which is necessary anyway as the functions are defined as the negative of what PS is expecting. However, this line of code would need to be manually altered every time the dimensionality of the function changes. Is there an alternative solution?

**Comment by ZacSpec** [comment]
I have a simple solution, if a bit messy, consisting of a bunch of if/elif statements.

**Comment by bstienen** [comment]
Let me reopen this issue. We plan om making (or at least: allowing for) some very high dimensional functions, so there should definitely be a more general solution. I will have a look at this soon.

**Comment by bstienen** [comment]
So I have looked into this and you are right that if/elif/else statements are messy. I personally don't quickly see an easy way out of this through your interface, so I have made a pull request with an update to the TestFunction and SimpleFunctionWrapper class that allows you to invert the function (i.e. multiply the evaluation result with -1). Simply call invert(True) on either the full or the wrapped function and function evaluations through the __call__ method will be multiplied by -1 automatically.
